server/api/orders.js

```diff
@@ -2,21 +2,79 @@ const router = require('express').Router();
 const {User, Order, Product, OrderProduct} = require('../db/models');
 const {Sequelize} = require('sequelize');
 
+// Get all cart items
+router.get('/', async (req, res, next) => {
+  try {
+    const id = req.user.id;
+    const notPurchasedOrder = await Order.findOne({
+      where: {
+        userId: id,
+        isPurchased: false
+      },
+      include: [
+        {
+          model: OrderProduct
+        }
+      ]
+    });
+    res.json(notPurchasedOrder);
+  } catch (e) {
+    next(e);
+  }
+});
+
+// Add to cart
 router.post('/', async (req, res, next) => {
   try {
     const pid = req.body.pid;
     const pqty = req.body.pqty;
 
     const product = await Product.findByPk(pid);
-    if (product.inventory > pqty) {
-      res.json(product);
+    if (product.inventory >= pqty) {
+      const isLoggedIn = req.user.id;
+
+      if (isLoggedIn) {
+        const cart = await Order.findOne({
+          where: {
+            userId: isLoggedIn,
+            isPurchased: false
+          }
+        });
+
+        if (cart) {
+          cart.grandTotal += product.price * pqty;
+          const newOrderItem = await OrderProduct.create({
```

Comment:
If the product exists in the cart, we want to update the existing entry rather than re-add / create the product in the cart again.

Reply:
Let's consider dropping the whole cart and re-populating from Redux/Local Storage?

Reply:
What about `Model.upsert()`?